lnwire/signature.go

```diff
@@ -13,23 +14,58 @@ import (
 // signature (raw bytes and *btcec.Signature).
 type Sig [64]byte
 
+var (
+	errSigTooShort = errors.New("malformed signature: too short")
+	errBadLength   = errors.New("malformed signature: bad length")
+	errBadRLength  = errors.New("malformed signature: bogus R length")
+	errBadSLength  = errors.New("malformed signature: bogus S length")
+	errRTooLong    = errors.New("R is over 32 bytes long without padding")
+	errSTooLong    = errors.New("S is over 32 bytes long without padding")
+)
+
 // NewSigFromRawSignature returns a Sig from a Bitcoin raw signature encoded in
 // the canonical DER encoding.
 func NewSigFromRawSignature(sig []byte) (Sig, error) {
 	var b Sig
 
-	if len(sig) == 0 {
-		return b, fmt.Errorf("cannot decode empty signature")
+	// Check the total length is above the minimal.
+	if len(sig) < btcec.MinSigLen {
+		return b, errSigTooShort
 	}
 
-	// Extract lengths of R and S. The DER representation is laid out as
-	// 0x30 <length> 0x02 <length r> r 0x02 <length s> s
-	// which means the length of R is the 4th byte and the length of S
-	// is the second byte after R ends. 0x02 signifies a length-prefixed,
+	// The DER representation is laid out as:
+	//   0x30 <length> 0x02 <length r> r 0x02 <length s> s
+	// which means the length of R is the 4th byte and the length of S is
+	// the second byte after R ends. 0x02 signifies a length-prefixed,
 	// zero-padded, big-endian bigint. 0x30 signifies a DER signature.
 	// See the Serialize() method for btcec.Signature for details.
-	rLen := sig[3]
-	sLen := sig[5+rLen]
+
+	// Reading <length>, remaining: [0x02 <length r> r 0x02 <length s> s]
+	sigLen := sig[1]
```

Comment:
nit: convert to int here? Also add empty line before comment below.

Reply:
Done!